<!-- register depth images to rgb images frame -->
<launch>
    <arg name="queue_size" default="5"/>
    <arg name="fill_upsampling_holes" default="true"/>
    <arg name="publish_pointcloud_xyz" default="false"/>
    <arg name="publish_pointcloud_xyzrgb" default="false"/>
    <arg name="aligned_depth_camera_info" default="aligned_depth/camera_info"/>
    <arg name="aligned_depth_image" default="aligned_depth/image_raw"/>
    <arg name="point_topic" default="depth/points"/>
    
    <!-- camera_left -->
    <include file="$(find instruct_to_policy)/launch/register_depth_single.launch">
        <arg name="camera_name" value="camera_left"/>
        <arg name="queue_size" value="$(arg queue_size)"/>
        <arg name="fill_upsampling_holes" value="$(arg fill_upsampling_holes)"/>
        <arg name="publish_pointcloud_xyz" value="$(arg publish_pointcloud_xyz)"/>
        <arg name="publish_pointcloud_xyzrgb" value="$(arg publish_pointcloud_xyzrgb)"/>
        <arg name="aligned_depth_camera_info" value="$(arg aligned_depth_camera_info)"/>
        <arg name="aligned_depth_image" value="$(arg aligned_depth_image)"/>
        <arg name="point_topic" value="$(arg point_topic)"/>
    </include>

    <!-- camera_top -->
    <include file="$(find instruct_to_policy)/launch/register_depth_single.launch">
        <arg name="camera_name" value="camera_top"/>
        <arg name="queue_size" value="$(arg queue_size)"/>
        <arg name="fill_upsampling_holes" value="$(arg fill_upsampling_holes)"/>
        <arg name="publish_pointcloud_xyz" value="$(arg publish_pointcloud_xyz)"/>
        <arg name="publish_pointcloud_xyzrgb" value="$(arg publish_pointcloud_xyzrgb)"/>
        <arg name="aligned_depth_camera_info" value="$(arg aligned_depth_camera_info)"/>
        <arg name="aligned_depth_image" value="$(arg aligned_depth_image)"/>
        <arg name="point_topic" value="$(arg point_topic)"/>
    </include>

    <!-- camer_right -->
    <include file="$(find instruct_to_policy)/launch/register_depth_single.launch">
        <arg name="camera_name" value="camera_right"/>
        <arg name="queue_size" value="$(arg queue_size)"/>
        <arg name="fill_upsampling_holes" value="$(arg fill_upsampling_holes)"/>
        <arg name="publish_pointcloud_xyz" value="$(arg publish_pointcloud_xyz)"/>
        <arg name="publish_pointcloud_xyzrgb" value="$(arg publish_pointcloud_xyzrgb)"/>
        <arg name="aligned_depth_camera_info" value="$(arg aligned_depth_camera_info)"/>
        <arg name="aligned_depth_image" value="$(arg aligned_depth_image)"/>
        <arg name="point_topic" value="$(arg point_topic)"/>
    </include>

    <!-- <include file="$(find instruct_to_policy)/launch/register_depth_single.launch">
        <arg name="camera_name" value="camera_front"/>
        <arg name="queue_size" value="$(arg queue_size)"/>
        <arg name="fill_upsampling_holes" value="$(arg fill_upsampling_holes)"/>
        <arg name="publish_pointcloud" value="$(arg publish_pointcloud)"/>
    </include> -->

</launch> 